scarb/src/bin/scarb/args.rs

```diff
@@ -120,14 +121,24 @@ impl ScarbArgs {
 
     /// Get [`ui::Verbosity`] out of these arguments.
     pub fn ui_verbosity(&self) -> scarb_ui::Verbosity {
-        let filter = self.verbose.log_level_filter().as_trace();
-        if filter >= LevelFilter::WARN {
-            scarb_ui::Verbosity::Verbose
-        } else if filter > LevelFilter::OFF {
-            scarb_ui::Verbosity::Normal
+        let verbosity = if !self.verbose.is_present() {
```

Comment:
I think we reached a point then `clap_verbosity_flag` just doesn't work for us. If we handled this env within clap argument, we wouldn't need a lot of the logic here.

Perhaps we should just copy-paste `clap_verbosity_flag` into `scarb_ui` and edit it to operate on UI Verbosity and tracing LogLevels at once? And support control via environment variables.